e2e/test/scenarios/filters-reproductions/dashboard-filters-reproductions.cy.spec.js

```diff
@@ -1782,14 +1782,22 @@ describe("issue 25374", () => {
   };
 
   beforeEach(() => {
-    cy.intercept("POST", "/api/card/*/query").as("cardQuery");
-    cy.intercept("POST", "/api/dashboard/*/dashcard/*/card//*/query").as(
-      "dashcardQuery",
-    );
+    // we are overriding the intercept set in e2e/support/helpers/e2e-misc-helpers.js
+    cy.intercept(
+      {
+        method: "GET",
+        url: "/api/dashboard/*?dashboard_load_id=**",
+      },
+      req => {
+        req.reply(res => {
+          res.body.last_used_param_values = {}; // remove param values to prevent random redirects caused by cached redirect
```

Comment:
we shouldn't override server behaviour in tests, should we? even if it's a flaky test, such change only hides the problem

Reply:
For the record the problem is that `last_used_param_values` are updated asynchronously in the backend which makes these kind of tests non-deterministic.
The solution is to use `changeSynchronousBatchUpdateSetting` util, see #49379.
Related Slack conversation: https://metaboat.slack.com/archives/C010L1Z4F9S/p1727947712609579